lib/jormungandr/test/unit/Cardano/Wallet/Jormungandr/BinarySpec.hs

```diff
@@ -76,90 +79,28 @@ import System.Directory
-                    \d501d0fa7e180d33987d17f77cbf70e1463bce01d32d952ed6f9823f0d\
-                    \69eb37e35f931417c6075e0f3e5858198fe15831ba7fb51368fa2f0ac2\
-                    \7a799032729e08a624a4aafb7a4dde35e4742d258d04c5f3ec87e616b9\
-                    \bcb0cdc070b503fe634b46010040a856b8a6f8d18d588b5e1cfd3ea2e5\
-                    \6ae45b80126bb25feb8ccde27fe61ebc7fd64deb7667ab1a79ca2448f5\
-                    \6e60f3097c2fa657febdec19e7bd7abfb0ea4705"
-            let block = Block
-                    BlockHeader
-                        { version = 0
-                        , contentSize = 458
-                        , slot = SlotId 0 0
-                        , chainLength = 0
-                        , contentHash = Hash $ unsafeFromHex
-                            "f7becdf807c706cef54ec4832d2a7475\
-                            \91c3f2141de3e4f2aef59a130d890c12"
-                        , headerHash = Hash $ unsafeFromHex
-                            "e1abfad5b57907832d22b219d2615b62\
-                            \55437765cfd993cd0ba31b0248bad464"
-                        , parentHeaderHash = Hash (BS.replicate 32 0)
-                        , producedBy = Nothing
-                        }
-                    [ Initial
-                        [ Block0Date (StartTime $ posixSecondsToUTCTime 1556202057)
-                        , Discrimination Testnet
-                        , Consensus BFT
-                        , SlotsPerEpoch $ W.EpochLength 2160
-                        , SlotDuration 15
-                        , EpochStabilityDepth (Quantity 10)
-                        , AddBftLeader $ LeaderId $ unsafeFromHex
-                            "30a694b80dbba2d1b8a4b55652b03d96\
-                            \315c8414b054fa737445ac2d2a865c76"
-                        , ConsensusGenesisPraosParamF (Milli 220)
-                        , MaxNumberOfTransactionsPerBlock 255
-                        , BftSlotsRatio (Milli 220)
-                        , AllowAccountCreation True
-                        , ConfigLinearFee $ LinearFee (Quantity 0) (Quantity 0)
-                        ]
-                    , Transaction (Tx
-                        { txid = Hash $ unsafeFromHex
-                            "6f5e01c34590f5ead789c234a816ac25\
-                            \41baece53f6e51bf52222bd7feb5cd04"
-                        , inputs = []
-                        , outputs =
-                            [ TxOut
-                                { address = unsafeDecodeAddress @'Testnet
-                                    "ta1svejfsmcd8qjy6y6xkghmaf6fu3f\
-                                    \fga99a59up04lrjnhpl8afzj7w4yyxw"
-                                , coin = Coin 14
-                                }
-                            ]
-                        }, [])
-                    , UnimplementedMessage 1
-                    , UnimplementedMessage 3
```

Comment:
I removed this tests mainly because `UnimplementedMessage 3` is not valid anymore. And I do not know how to generate binary with `UnimplementedMessage 4` frankly speaking

Reply:
managed to produce binary with Transaction and UnimplementedMessage

Reply:
worth to be noted. I changed genesis.yaml in such a way that from `initial` only this stayed : 
```
initial:
  # Funds & certificates for a single stake pool.
  # https://input-output-hk.github.io/jormungandr/advanced/03_starting_genesis_praos_blockchain.html
  #
  # (The faucet-addresses above don't delegate.)
  #
  # An account address:
  - fund:
      - address: ca1skrhlwezswapa4tdsd0m3ntxd99ggqmqu6wxjzsy4mhnjc5umkqy7pccwfa
        value: 1
  # Stake pool registration certificate:
  - cert: cert1qvqqqqqqqqqqqqqqqqqqq0p5avfqqqqqqqqqqqqqqqqsqqty8vgjhtxtlv3f3nwjuqk756syldxq0yw06chzda778taqw0pgguqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqpqqqqqqqqqqqqq3nxuq30jc00eqk9q7nmse2t2unad302gza5gspqqjuazl0qaffv5s2hdy3dtjsg0jza7n6a9h5yg2pfj00n2dx5tu9e09wmms7duu7srkfye0
  # This certificate delegates the stake at the account address to the stake pool
  - cert: cert1qxrhlwezswapa4tdsd0m3ntxd99ggqmqu6wxjzsy4mhnjc5umkqylgy6urdpucvwavy70dudw03xttcclp7565eq8pht7q34748v6q68cvr6s5
```
and produced block0. We do not have anything other than Transaction and UnimplementedMessage

Reply:
I do not know the answer for this and unfortunately cannot check against binary at this moment

Reply:
With the following patch:
```patch
diff --git a/lib/jormungandr/src/Cardano/Wallet/Jormungandr/Binary.hs b/lib/jormungandr/src/Cardano/Wallet/Jormungandr/Binary.hs
index a7fff9b5d..977df5574 100644
--- a/lib/jormungandr/src/Cardano/Wallet/Jormungandr/Binary.hs
+++ b/lib/jormungandr/src/Cardano/Wallet/Jormungandr/Binary.hs
@@ -299,8 +299,8 @@ getMessage = label "getMessage" $ do
         0 -> Initial <$> getInitial
         1 -> unimpl
         2 -> Transaction <$> getTransaction fragId
-        3 -> TransactionWithDelegation <$> getDelegatedCertificateTransaction fragId
-        4 -> unimpl
+        3 -> unimpl
+        4 -> TransactionWithDelegation <$> getDelegatedCertificateTransaction fragId
         5 -> unimpl
         other -> fail $ "Unexpected content type tag " ++ show other

@@ -348,10 +348,10 @@ legacyUtxoWitness xpub bytes = TxWitness $ BL.toStrict $ runPut $ do
 getDelegatedCertificateTransaction
     :: Hash "Tx"
     -> Get (PoolId, ByteString, Tx, [TxWitness])
-getDelegatedCertificateTransaction tid = do
-    (tx, wits) <- getGenericTransaction tid
+getDelegatedCertificateTransaction tid = label "getDelegatedCertificateTransaction" $ do
     accId <- getByteString 32
     poolId <- getByteString 32
+    (tx, wits) <- getGenericTransaction tid
     pure (PoolId poolId, accId, tx, wits )

 putStakeCertificate
```

(from following @KtorZ's PR, since I assume he checked the rust-code)

I get a promising failure in the test:
<img width="530" alt="Skärmavbild 2019-10-31 kl  11 00 48" src="https://user-images.githubusercontent.com/304423/67937777-75d15a80-fbce-11e9-84c7-eeaeaf7dd668.png">

i.e.

- I believe we most certainly can test this
- I believe the decoder is currently wrong (the payload should be parsed before the IOs)